tests/functional/TicketTask.php

```diff
@@ -683,4 +689,373 @@ public function testParentMetaSearchOptions()
+        $task_right = $this->createItem(ProfileRight::class, [
+            'profiles_id' => $profile->getID(),
+            'name' => 'task',
+            'rights' => CommonITILTask::ADDMY,
+        ]);
+
+        $this->login();
+
+        $this->boolean((new \TicketTask())->canCreate())->isTrue();
+
+        // Check that the last_rights_update field is not null
+        $this->variable($profile->fields['last_rights_update'])->isNotNull();
+
+        $profile->getFromDB($profile->getID());
+
+        $this->login();
+
+        // Get a ticket
+        $ticket = self::getNewTicket();
+
+        // Create a task
+        $task = $this->createItem(
+            \TicketTask::class,
+            [
+                'state' => \Planning::TODO,
+                'tickets_id' => $ticket,
+                'content' => "Task",
+                'users_id' => $uid,
+            ]
+        );
+
+        $this->boolean((new \TicketTask())->canUpdateItem())->isFalse();
+
+        // Create a profile right to add a task to the ticket
+        $this->updateItem(
+            ProfileRight::class,
+            $task_right->getID(),
+            [
+                'rights' => CommonITILTask::UPDATEMY,
+            ]
+        );
+
+        $this->login();
+
+        $this->boolean($task->canUpdateItem())->isTrue();
+    }
+
+    /**
+     * Test that the user can create a task in function of his status
+     *
+     * @param string $profileName
+     * @param integer $right
+     * @param string $ticketUserType
+     *
+     * @return void
+     */
+    private function createTaskAndAssertCanCreate($profileName, $right, $ticketUserType)
+    {
+        global $DB;
+
+        $this->login();
+        $uid = getItemByTypeName(User::class, TU_USER, true);
+
+        //Create profile
+        $profile = $this->createItem(Profile::class, [
+            'name' => $profileName,
+            'interface' => 'central',
+        ]);
+
+        $profile_user = new Profile_User();
+        $profile_user->getFromDBByCrit([
+            'users_id' => $uid,
+        ]);
+
+        $DB->delete(
+            ProfileRight::getTable(),
+            [
+                'profiles_id' => $profile->getID(),
+                'name' => 'task'
+            ],
+        );
+
+        //Add the profile to the user
+        $DB->update(
+            Profile_User::getTable(),
+            ['profiles_id' => $profile->getID()],
+            ['users_id' => $uid],
+        );
+
+        $this->login();
+
+        $ticket = self::getNewTicket();
+
+        $this->boolean((new \TicketTask())->canCreate())->isFalse();
+
+        $task = $this->createItem(
+            \TicketTask::class,
+            [
+                'state' => \Planning::TODO,
+                'tickets_id' => $ticket,
+                'content' => "Task",
+                'users_id' => $uid,
+            ]
+        );
+
+        $this->boolean($task->canCreateItem())->isFalse();
```

Comment:
It seems weird to create an item and then check if it can be created. It indeed checks the method logic, but it does not seems logical when reading the test.

Same remark for other tests on this method.

```suggestion
        $task = new \TicketTask();
        $task->fields = [
            'state' => \Planning::TODO,
            'tickets_id' => $ticket,
            'content' => "Task",
            'users_id' => $uid,
        ];

        $this->boolean($task->canCreateItem())->isFalse();
```